### PR TITLE
Cleaning up the woeful zexcon

### PR DESCRIPTION
**Pull request by lavillastrangiato**
## About The Pull Request

Ports Ratwood's sexcon, without the zape, fetish content, a few extraneous eyebrow-raising stuff, and makes wording less explicit. **Edit 18/12/2024:** Adds the GENITAL-MINCING CODE, by https://github.com/Darkrp-community/OpenKeep/pull/1254/files, to this PR to cut down on the merge conflicts.

- I fixed the pink screen overlays!
- Some same-sex acts have been added.
- There are speeds (which technically already existed) and forces now. Highly immersive. (I have removed the "brutal" and "unrelenting" speeds and forces because that's just too much)
- Religious roles have had their virginity stress events rejigged a bit to be *lore-accurate*. If you are a religious role, the virginity-loss stress event will only proc if your patron is Astrata, Noc, Malum, or the Forgotten God, and you are not married.
- Adminlogs now exist for: trying to initiate sex with an unconscious person, trying to initiate sex with a dead person (or a disconnected person), and trying to initiate sex with someone in combat mode. In all cases, you can't open the menu and a notification will pop up in chat as to why you can't do it.
image

## Why It's Good For The Game

- The original sexcon is poorly coded. It is barely documented, unnecessarily long and yet oddly barebones while also lacking atomisation. It is very difficult to remove stuff like the zapecode or even shorten an if-statement without causing problems.
- This sexcon is better-atomised and better-documented. It is also easier on the eyes as a piece of code.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/datums/sexcon/sex_actions/deviant/frotting.dm
Comment (BadAtThisGame302) [inline code/datums/sexcon/sex_actions/deviant/frotting.dm:1]: Frotting... oh my God... it's so peak.

**Comment by bowlofcereal** [comment]
https://github.com/user-attachments/assets/9c3abb56-9236-401b-80a3-4763ce7bdf78

**Comment by messagecypher** [comment]
thanks lvs

**Comment by NPC1314** [comment]
Not a fan of menus popping up, but then again I'm not going to try to do any work with this system and I'm not testing it either so unless anyone got better ideas I guess we go with this, since the old was broken by #1145 I'm told.

File: code/datums/sexcon/sexcon.dm [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+	virginity = FALSE
+	if(mind)
+		switch(mind.assigned_role)
+			if("Priest")
+				P.add_stress(/datum/stressevent/virginchurch)
+			if("Acolyte")
+				P.add_stress(/datum/stressevent/virginchurch)
+			if("Cleric")
+				P.add_stress(/datum/stressevent/virginchurch)
+			if("Paladin")
```
Comment (BadAtThisGame302) [inline code/datums/sexcon/sexcon.dm:464]: You forgot to add Templar to the list of Church virgin roles.

Comment (lavillastrangiato) [reply code/datums/sexcon/sexcon.dm:464]: deliberate because i thought they weren't virgins but they are

**Comment by ThePainkiller** [comment]
Wondering if there's any kinks (pun intended) left to iron on this PR or if it's finally time to give this a full merge?

**Review by NPC1314 — APPROVED** [review]

**Pull request merged**